src/main/java/org/folio/inventory/resources/AbstractInstances.java

```diff
@@ -76,12 +85,66 @@ protected CompletableFuture<List<Response>> updateInstanceRelationships(Instance
     relatedInstancesClient.getAll(query, future::complete);
 
     return future.thenCompose(result ->
-      updateInstanceRelationships(instance, relatedInstancesRepository, result));
+      updateRelatedInstances(instance, relatedInstancesRepository, result));
   }
 
-  private CompletableFuture<List<Response>> updateInstanceRelationships(Instance instance,
+  private CompletableFuture<List<Response>> updateRelatedInstances(Instance instance,
     CollectionResourceRepository relatedInstancesClient, Response result) {
 
+    JsonObject json = result.getJson();
+
+    List<JsonObject> relatedInstancesList = JsonArrayHelper.toList(json.getJsonArray(Instance.RELATED_INSTANCES_KEY));
+    Map<String, RelatedInstance> existingRelatedInstances = new HashMap<>();
+
+    relatedInstancesList.stream().map(RelatedInstance::new).forEachOrdered(related ->
+      existingRelatedInstances.put(related.id, related)
+    );
+
+    Map<String, RelatedInstance> updatingRelatedInstances = new HashMap<>();
+    if (instance.getRelatedInstances() != null) {
+      instance.getRelatedInstances().forEach(updating -> {
+        String id = (updating.id == null ? UUID.randomUUID().toString() : updating.id);
```

Comment:
I'm not sure this will work for related instances in the current form. 

At the moment the `id` of the relationship isn't in the business logic API representation. If the client abides by that, then the `id` received from the client will always be null. Which might mean that a new related instance record is created.

This might be why the instance relationships representation exposes an ID (which unfortunately, has some negative impacts too (refer to [MODINV-674](https://issues.folio.org/browse/MODINV-674?focusedCommentId=128833&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-128833) for more details).

Reply:
How should we address this issue? Should it also be addressed for instance relationships at this time?

Reply:
> How should we address this issue? 

If it's possible to hide the `id` of the related instance records (which are an artefact of the storage / business logic split rather than the domain) then I think we should.

This will mean the logic needed to detect how a related instance translates to storage will need to be different from instance relationships.

> Should it also be addressed for instance relationships at this time?

I don't think so. This would be a breaking change and folks haven't decided what to do about this. I don't think we should mix these two pieces of work together.